nitro-protocol/test/contracts/LedgerFinancingApp/requireStateSupported.test.ts

```diff
@@ -0,0 +1,291 @@
+  // 1/100 -> 1% daily percentage yield.
+  dpyNum: 1,
+  dpyDen: 100,
+  blocknumber: 1,
+  principal: {
+    asset: [MAGIC_NATIVE_ASSET_ADDRESS],
+    amount: [500],
+  },
+  collectedInterest: {
+    asset: [MAGIC_NATIVE_ASSET_ADDRESS],
+    amount: [0],
+  },
+};
+
+const baseState: State = {
+  turnNum: 0,
+  isFinal: false, // intermediary wants to force finalization
+  channelNonce: '0x8',
+  participants,
+  challengeDuration,
+  outcome: initialOutcome,
+  appData: appDataABIEncode(baseAppData),
+  appDefinition: APPDEF,
+};
+
+const variablePart = getVariablePart(baseState);
+
+const signedByMerchant: RecoveredVariablePart = {
+  variablePart,
+  signedBy: BigNumber.from(0b10).toHexString(),
+};
+const signedByIntermediary: RecoveredVariablePart = {
+  variablePart,
+  signedBy: BigNumber.from(0b01).toHexString(),
+};
+const signedByBoth: RecoveredVariablePart = {
+  variablePart,
+  signedBy: BigNumber.from(0b11).toHexString(),
+};
+
+const fixedPart = getFixedPart(baseState);
+
+beforeAll(async () => {
+  ledgerFinancingApp = setupContract(provider, LedgerFinancingAppArtifact, APPDEF);
+});
+
+describe('requireStateSupported', () => {
+  it('accepts unanimous states', async () => {
+    await ledgerFinancingApp.requireStateSupported(fixedPart, [], signedByBoth);
+  });
+
+  it('accepts legitimate interest calculations', async () => {
+    // construct a proof+candidate test case with fair interest calculation, assert passing    // with the interest rate applied.
+    // test case:
+    // - appdata interest rate is 1% per day
+    // - initial outcome is 500:500, with 500 principal
+    // - 1 day passes
+    // - challenge outcome is 505:495
+    advanceOneDay();
+
+    const challengeState: State = {
+      ...baseState,
+      turnNum: baseState.turnNum + 1,
+      outcome: computeOutcome({
+        [MAGIC_NATIVE_ASSET_ADDRESS]: {[intermediary]: 505, [merchant]: 495}, // intermediary picks up 1% of the principal
+      }),
+    };
+    const challengeWithIntermediarySignature: RecoveredVariablePart = {
+      variablePart: getVariablePart(challengeState),
+      signedBy: BigNumber.from(0b01).toHexString(),
+    };
+
+    await ledgerFinancingApp.requireStateSupported(
+      fixedPart,
+      [signedByBoth],
+      challengeWithIntermediarySignature
+    );
+  });
+
+  it('reverts if the proof block number is in the future', async () => {
+    // create a challenge where the intermediary fakes the blocknumber
+    // (in order to get a higher interest rate).
+    const futureState: State = {
+      ...baseState,
+      appData: appDataABIEncode({...baseAppData, blocknumber: baseAppData.blocknumber + 1000000}),
+    };
+    const signedByIntermediary: RecoveredVariablePart = {
+      variablePart: getVariablePart(futureState),
+      signedBy: BigNumber.from(0b01).toHexString(),
+    };
+    await expectRevert(() =>
+      ledgerFinancingApp.requireStateSupported(fixedPart, [signedByBoth], signedByIntermediary)
+    );
```

Comment:
Where does this trigger a revert? I think it would be good to attach a named error / revert string to this case and assert on that.

Reply:
76f8f179

Test case is removed.